blockchain/blockchain.go

```diff
@@ -582,9 +582,6 @@ func (bc *BlockChain) setHeadBeyondRoot(head uint64, root common.Hash, repair bo
 					newHeadBlock = bc.GetBlock(newHeadBlock.ParentHash(), newHeadBlock.NumberU64()-1) // Keep rewinding
 				}
 			}
-			if newHeadBlock.NumberU64() == 0 {
```

Comment:
I think it's better to check the state at genesis rather than completely remove this condition.

Reply:
I mean it's better to check if the genesis block really has a state.

```
if newHeadBlock.NumberU64() == 0 && noStateAtGenesis {
  return 0, errors.New("rewound to block number 0, but repair failed")
}

```

Reply:
It has already been checked here:
```
if _, err := state.New(newHeadBlock.Root(), bc.stateCache, bc.snaps, nil); err != nil {

```
And if `newHeadBlock = bc.genesisBlock` applies, does it mean exiting the state?

Reply:
Okay, let me apply them
```
if newHeadBlock.NumberU64() == 0 {
	if _, err := state.New(newHeadBlock.Root(), bc.stateCache, bc.snaps, nil); err != nil {
		return 0, errors.New("rewound to block number 0, but repair failed")
	}
}
```